### PR TITLE
Example does not work; brew install too invasive

### PR DESCRIPTION
**Pull request by jmcgeheeiv**
Thank you for extending this module to work with Cognito and publishing it. While this is a fine start, I could not get it to work. This pull request contains my files at the point where I gave up trying to make it work with Cognito.

This pull request just illustrates the issues I found; **it is not intended to be merged as-is.**

Here are the issues I found:
- The example is incomplete. It does not specify required variables.
- I am on Ubuntu 18.04 and I do not care to install `brew`
- I think it is inappropriate for Terraform to install packages and modify user configuration
- `cloudfront_auth_brach` is misspelled
- I am unsure of my assumptions about the nodejs version--I did not get that far
- In `variables.tf` I commented on variables that needed more explanation

Let me know if you need anything more from me.

### CODE REVIEW

File: example/main.tf [hunk 1]
```diff
@@ -2,14 +2,19 @@ module "cloudfront_auth" {
   source = "../"
 
   auth_vendor             = "cognito"
+  authz                   = "2"  # Is this a number or a name?
```
Comment (mslipets) [inline example/main.tf:5]: As stated in original description at [variables.tf](https://github.com/scalefactory/terraform-cloudfront-auth/blob/283c2c9179dec049e2946fc3a58b7014b5289c44/variables.tf#L41)
this is applicable only for microsoft or google auth_vendor (defines additional params )

File: example/main.tf [hunk 2]
```diff
@@ -2,14 +2,19 @@ module "cloudfront_auth" {
...
   cloudfront_distribution = "private.example.com"
+  cloudfront_aliases      = ["secret.example.com"]
+  base_uri                = "https://auth-xxxxxx.auth.us-east-1.amazoncognito.com"
```
Comment (mslipets) [inline example/main.tf:8]: This is the base uri for congnito [idp](https://github.com/mslipets/terraform-cloudfront-auth/blob/8b28e4b72f88ce62f87657e8f74067ff48a828c1/variables.tf#L29) settings endpoint
see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-app-idp-settings.html

File: example/main.tf [hunk 3]
```diff
@@ -2,14 +2,19 @@ module "cloudfront_auth" {
...
   client_id               = "CHANGE_ME"
   client_secret           = "CHANGE_ME"
+  nodejs_version          = "12.18.0"
   redirect_uri            = "https://private.example.com/callback"
-  github_organization     = "exampleorg"
 
   bucket_name                    = "private.example.com"
   region                         = "eu-west-1"
+  # TODO: probably wrong
+  bucket_access_roles_arn_list   = []
```
Comment (mslipets) [inline example/main.tf:17]: "A (list of) Role ARN which granted RW rights to bucket (to be used by instance profiles in order to upload/cleanup bucket content)"

File: variables.tf [hunk 4]
```diff
@@ -11,7 +11,7 @@ variable "auth_vendor" {
 
 variable "cloudfront_distribution" {
   type        = string
-  description = "The cloudfront distribtion"
+  description = "The cloudfront distribution"   # URL?
```
Comment (mslipets) [inline variables.tf:14]: not quite URL, rather CNAME for CF distribution concatenated with rest of the [aliases ](https://github.com/mslipets/terraform-cloudfront-auth/blob/8b28e4b72f88ce62f87657e8f74067ff48a828c1/main.tf#L191)

File: variables.tf [hunk 5]
```diff
@@ -31,7 +31,7 @@ variable "base_uri" {
 
 variable "redirect_uri" {
   type        = string
-  description = "The redirect uri "
+  description = "The redirect uri" # redirect to what?
```
Comment (mslipets) [inline variables.tf:34]: Essentially to CF SPA/requested page,
redirect uri for [oauth callback](https://www.oauth.com/oauth2-servers/redirect-uris/)
e.g.: "https://${distribution_url}/_callback" (which is handled by [Widen/cloudfront-auth](https://github.com/Widen/cloudfront-auth/wiki/AWS-Cognito-Configuration))

File: variables.tf [hunk 6]
```diff
@@ -50,7 +50,7 @@ variable "authz" {
   type        = string
   default     = "1"
   description = "The authorisation method (google, microsoft only). Mirosoft: (1) Azure AD Login (default)\n   (2) JSON Username Lookup\n\n Google: (1) Hosted Domain - verify email's domain matches that of the given hosted domain\n   (2) HTTP Email Lookup - verify email exists in JSON array located at given HTTP endpoint\n   (3) Google Groups Lookup - verify email exists in one of given Google Groups"
-}
+} # Is this a string containing a number? Numbers 1 and 2 are repeated in the description.
```
Comment (mslipets) [inline variables.tf]: Yes, this was leaved as is in [original implementation](https://github.com/scalefactory/terraform-cloudfront-auth/blob/fe68938cdb7eebe5c56758d3ed4f51426d548d0f/variables.tf#L38), specific for [(google, microsoft only)](https://github.com/scalefactory/terraform-cloudfront-auth/blob/fe68938cdb7eebe5c56758d3ed4f51426d548d0f/docs/terraform.md)
set to default value solely to satisfy parameter presence requirement for [build.js](https://github.com/mslipets/terraform-cloudfront-auth/blob/8b28e4b72f88ce62f87657e8f74067ff48a828c1/build.js#L212)

File: variables.tf [hunk 7]
```diff
@@ -94,13 +94,13 @@ variable "cloudfront_default_root_object" {
...
   description = "ACM Certificate ARN for Cloudfront"
-  default     = ""
+  default     = "" # Perhaps this should be required
 }
 
 variable "nodejs_version" {
   type        = string
   default     = "10.19.0"
-  description = "Node.js runtime version."
+  description = "Node.js runtime version." # For building or for running?
```
Comment (mslipets) [inline variables.tf:113]: both =)

**Review by mslipets — COMMENTED** [review]
RE> This pull request just illustrates the issues I found; **it is not intended to be merged as-is.**

@jmcgeheeiv please review following [PR#4](https://github.com/mslipets/terraform-cloudfront-auth/pull/4) for some adjustment's you've suggested.

File: variables.tf (hunk 4 above)
Comment (jmcgeheeiv) [reply variables.tf:14]: I do not see the description updated in PR #4. I do not understand your explanation.

Don't forget to update the spelling of "distribution."

File: variables.tf (hunk 5 above)
Comment (jmcgeheeiv) [reply variables.tf:34]: I do not see this updated in PR #4. I suggest this description:
> URL for the authorization server to call after authentication is successfully completed.

File: variables.tf (hunk 7 above)
Comment (jmcgeheeiv) [reply variables.tf:113]: I do not see this updated in PR #4.

**Pull request closed without merging**